### PR TITLE
Packed

### PR DESCRIPTION
**Pull request by adam-coogan**
Added `Packed` dummy class, renamed `x_to_dict` to `pack` and updated tests. Also added commented-out sketch of the decorator @ConnorStoneAstro suggested for toggling between `x` and passing args/kwargs.

### CODE REVIEW

**Comment by adam-coogan** [comment]
(Oops, I accidentally pushed the new interpolator to this branch instead of making a new PR...)

The latest commit adds `caustic.utils.interp2d`, loosely based off [this gist](https://gist.github.com/peteflorence/a1da2c759ca1ac2b74af9a83f69ce20e). Key features:
- Fully compatible with automatic differentiation and `vmap`!
- Similar to `torch.nn.functional.grid_sample` with `align_corners=False`, except that the input tensor must be 2D.
- Supports nearest-neighbor and bilinear interpolation
- Supports padding with zeros or extrapolation
I've updated `KappaGrid`, `ImageSource` and the relevant tests.

**Review by AlexandreAdam — APPROVED** [review]

File: src/caustic/parametrized.py [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
 import torch
 from torch import Tensor
 
+from .packed import Packed
 from .parameter import Parameter
 
 __all__ = ("Parametrized",)
 
 
+# def unpack_method(n_leading_args=0):
```
Comment (ConnorStoneAstro) [inline src/caustic/parametrized.py:16]: remove these comments?

File: src/caustic/lenses/kappa_grid.py [hunk 2]
```diff
@@ -36,6 +36,15 @@ def __init__(
         """
         super().__init__(name, cosmology, z_l)
 
+        if kappa_map is not None and kappa_map.ndim != 2:
+            raise ValueError(
+                f"kappa_map must be 2D (received {kappa_map.shape}D tensor)"
```
[the remark this answers is not in the document]
Comment (adam-coogan) [reply src/caustic/lenses/kappa_grid.py:41]: Good catch! I meant to write `kappa_map.ndim`.

File: src/caustic/lenses/kappa_grid.py [hunk 3]
```diff
@@ -113,34 +122,34 @@ def mode(self, mode):
 
         self._mode = mode
 
-    def _check_kappa_map_shape(self, kappa_map):
-        if kappa_map.ndim != 4:
-            raise ValueError("kappa map must have four dimensions")
+    # def _check_kappa_map_shape(self, kappa_map):
```
[the remark this answers is not in the document]
Comment (adam-coogan) [reply src/caustic/lenses/kappa_grid.py:125]: Yes, will do

File: src/caustic/parametrized.py (hunk 1 above)
Comment (adam-coogan) [reply src/caustic/parametrized.py:16]: Will do

**Comment by ConnorStoneAstro** [comment]
The unit tests are still failing, could you try updating the "requirements.txt" file?

**Review by ConnorStoneAstro — APPROVED** [review]
